addons/mail/models/res_partner.py

```diff
@@ -45,11 +45,10 @@ def _get_needaction_count(self):
     def _get_starred_count(self):
         """ compute the number of starred of the current partner """
         self.ensure_one()
-        self.env.cr.execute("""
-            SELECT count(*) as starred_count
-            FROM mail_message_res_partner_starred_rel R
-            WHERE R.res_partner_id = %s """, (self.id,))
-        return self.env.cr.dictfetchall()[0].get('starred_count')
+        return self.env['mail.message'].with_user(self.user_ids[0]).search([
+            ('starred', '=', True),
+            '|', '|', ('model', '!=', 'mail.channel'), ('attachment_ids', '!=', False), ('body', '!=', '')
```

Comment:
Why is there a condition on `model`?

For the answer to the other questions, please check the JS methods I sent you in a previous message, they should contain all you need to know about the condition you need for properly detecting empty, including the various format existing for empty `body` itself.

To have a tracking message you just have to change a tracked field on any form view.

And for subtype description look for `model="mail.message.subtype"` in data xml which have a `name="description"`. Some of them are posted without content and rely on that description.

Reply:
Note it would probably be easier if we made a new computed field, or at least a method returning the domain, for "is empty" on message because we'll need to make the same kind of fix in a lot of other places (inbox counter/needaction counter, channel unseen/unread, ...).

Reply:
Hi @seb-odoo , thanks again for your review
I just pushed the changes taking your comments into account, it should be good now :slightly_smiling_face:

Reply:
Thank you for the update, it looks much better now.

You'll need to adapt test (see runbot), and also it would be nice to add tests covering each possible case for the new compute by itself, as well as other tests covering the star counter in the different cases.

Reply:
Hi, for ci/template this happen because a field is added (that is not yet in saas-15.1 but will be in the forward-port), you need an exemption of the test from xdo before a merge can happen.